src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -265,73 +392,104 @@ public T get(E event) {
+			return getEventValueGetter(event, type, 0, false);
 		return null;
 	}
 
 	/**
 	 * Check if the event value states to exclude events.
 	 * 
-	 * @param ev
-	 * @param e
+	 * @param eventValueInfo
+	 * @param event
 	 * @return boolean if true the event value passes for the events.
 	 */
-	@SuppressWarnings("unchecked")
-	private static boolean checkExcludes(EventValueInfo<?, ?> ev, Class<? extends Event> e) {
-		if (ev.excludes == null)
+	private static boolean checkExcludes(EventValueInfo<?, ?> eventValueInfo, Class<? extends Event> event) {
+		if (eventValueInfo.excludes == null)
 			return true;
-		for (Class<? extends Event> ex : (Class<? extends Event>[]) ev.excludes) {
-			if (ex.isAssignableFrom(e)) {
-				Skript.error(ev.excludeErrorMessage);
+		for (Class<? extends Event> ex : (Class<? extends Event>[]) eventValueInfo.excludes) {
+			if (ex.isAssignableFrom(event)) {
+				Skript.error(eventValueInfo.excludeErrorMessage);
 				return false;
 			}
 		}
 		return true;
 	}
-	
+
 	@Nullable
-	private static <E extends Event, F, T> Getter<? extends T, ? super E> getConvertedGetter(EventValueInfo<E, F> i, Class<T> to, boolean checkInstanceOf) {
-		Converter<? super F, ? extends T> converter = Converters.getConverter(i.c, to);
+	private static <E extends Event, F, T> Getter<? extends T, ? super E> getConvertedGetter(EventValueInfo<E, F> eventValueInfo, Class<T> to, boolean checkInstanceOf) {
+		Converter<? super F, ? extends T> converter = Converters.getConverter(eventValueInfo.type, to);
 		if (converter == null)
 			return null;
 		return new Getter<T, E>() {
 			@Override
 			@Nullable
 			public T get(E e) {
-				if (checkInstanceOf && !i.event.isInstance(e))
+				if (checkInstanceOf && !eventValueInfo.event.isInstance(e))
 					return null;
-				F f = i.getter.get(e);
+				F f = eventValueInfo.getter.get(e);
 				if (f == null)
 					return null;
 				return converter.convert(f);
 			}
 		};
 	}
-	
-	public static boolean doesEventValueHaveTimeStates(Class<? extends Event> e, Class<?> c) {
-		return getEventValueGetter(e, c, -1, false) != null || getEventValueGetter(e, c, 1, false) != null;
+
+	/**
+	 * Checks if an event value from the defined event class and return type has any time states that aren't present.
+	 * 
+	 * @param event the event to check event values for.
+	 * @param type the return type of the event value to search for.
+	 * @return boolean if any event values meet the parameters and have a time state that isn't present.
+	 */
+	public static boolean doesEventValueHaveTimeStates(Class<? extends Event> event, Class<?> type) {
+		return getEventValueGetter(event, type, TIME_PAST, false) != null || getEventValueGetter(event, type, TIME_FUTURE, false) != null;
 	}
-	
+
+	/**
+	 * Prints a list of registered event values to console if verbose is 'debug'
+	 */
+	public static void debug() {
+		StringBuilder builder = new StringBuilder("All registered event values in order\nPast:");
```

Comment:
I would add in a check to just stop if Skript isn't in debug mode

So that the message isn't created

Reply:
It already has one when it's finished registering the event values. This is also for addons to debug as addons are a large majority of event-values too.